epoch_publisher/src/client.rs

```diff
@@ -0,0 +1,149 @@
+use std::{collections::HashMap, sync::Arc};
+
+use crate::error::Error;
+use bytes::Bytes;
+use common::{host_info::HostInfo, network::fast_network::FastNetwork};
+use flatbuf::epoch_publisher_flatbuffers::epoch_publisher::*;
+use flatbuffers::FlatBufferBuilder;
+use tokio::sync::{oneshot, Mutex};
+use tokio_util::sync::CancellationToken;
+use uuid::Uuid;
+
+/// Provides an async rpc interface to a specific epoch publisher.
+pub struct EpochPublisherClient {
+    fast_network: Arc<dyn FastNetwork>,
+    range_server_info: HostInfo,
+    // TODO: make more typeful and store more information to e.g. allow timing out.
+    outstanding_requests: Mutex<HashMap<Uuid, oneshot::Sender<Bytes>>>,
```

Comment:
What are the keys?

Reply:
Also consider something like the Dashmap crate - supposed to be much faster if you expect a lot of read-writes to this map. It would be O(transactions) on the client, right?

Reply:
Keys are request ids. I'll look itno Dashmap